website/package.json

```diff
@@ -87,6 +87,7 @@
     "@docusaurus/tsconfig": "3.0.0-beta.0",
     "@types/jest": "^29.5.3",
     "cross-env": "^7.0.3",
-    "rimraf": "^3.0.2"
+    "rimraf": "^3.0.2",
+    "search-insights": "^2.8.2"
```

Comment:
@shortcuts FYI this requires me to add this extra deps to our website because it's an optional peerDeps of docsearch react but it's used for typechecking. Without this extra deps we still get the TypeScript errors:

```
Error: ../node_modules/@algolia/autocomplete-plugin-algolia-insights/dist/esm/types/InsightsClient.d.ts(1,75): error TS2307: Cannot find module 'search-insights' or its corresponding type declarations.
Error: ../node_modules/@algolia/autocomplete-plugin-algolia-insights/dist/esm/types/InsightsClient.d.ts(2,212): error TS2307: Cannot find module 'search-insights' or its corresponding type declarations.
```

This is due to our usage of TypeScript `"skipLibCheck": false,`


Not sure why this deps is an optional peer deps, or if you should do anything, but I thought you might be interested to know ;)

Reply:
hey @slorber, there's a new optional feature with DocSearch which allow users to track events and we use the search-insights Algolia library to do so, I believe it's optional because the option is opt-in, so I guess the behavior is correct here?

Reply:
![CleanShot 2023-09-19 at 10 47 52@2x](https://github.com/facebook/docusaurus/assets/749374/53409710-7c70-444a-a6bc-ffe8392eade0)

@shortcuts if I use `@algolia/autocomplete-plugin-algolia-insights`, isn't it expected that `search-insights` should be available in deps? Maybe it's `@algolia/autocomplete-plugin-algolia-insights` that should be a peer deps instead?

I don't have a solution to this problem, and not sure to understand it fully, but something feels a bit weird to me in your current setup 😅 Even if your package does not require/import 'search-insights', it still depends on its types, so types should rather be included? I don't know. We probably won't be the only one using `skipLibCheck: false` and stumble upon this issue.